File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 9/14/2025, 12:54:18 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        minP= math.inf
        for p in prices:
            minP = min(p, minP)
            maxP= max(p-minP, maxP)
        return maxP

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.c
// Last updated: 9/14/2025, 12:54:23 PM
int maxProfit(int* prices, int pricesSize) {
    int prof = 0;
    int min = prices[0];
    for (int i = 0; i < pricesSize; i++) {
        if(prices[i]< min) min = prices[i];
        prof = prices[i]-min > prof ? prices[i]-min : prof;
    }
    return prof;
}